python/pyarrow/tests/test_dataset.py

```diff
@@ -1605,3 +1605,21 @@ def test_dataset_schema_metadata(tempdir):
     assert b"pandas" in schema.metadata
     # ensure it is still there in a projected schema (with column selection)
     assert schema.equals(projected_schema, check_metadata=True)
+
+
+@pytest.mark.parquet
+@pytest.mark.pandas
+def test_dataset_project_only_partition_columns(tempdir):
+    # ARROW-8729
+    import pyarrow.parquet as pq
+
+    table = pa.table({'part': 'a a b b'.split(), 'col': list(range(4))})
+
+    path = str(tempdir / 'test_dataset')
+    pq.write_to_dataset(table, path, partition_cols=['part'])
+    dataset = ds.dataset(path, partitioning='hive')
+
+    all_cols = dataset.to_table(use_threads=False)
+    part_only = dataset.to_table(columns=['part'], use_threads=False)
+
+    assert all_cols.column('part') == part_only.column('part')
```

Comment:
```suggestion
    assert all_cols.column('part').equals(part_only.column('part'))
```

`==` is now element-wise, and the "truthyness" of a boolean chunked array is always True, regardless of the values in it .. (so as long both tables have a "part" column, the above check would always return True even if the columns itself are not equal)